src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -368,153 +373,151 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
+                        LuceneConfig.SearchNameWeight : (LuceneConfig.WildcardsSearchNameWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.NameSplitted):
                     //Under this case the NameSplitted field will have less weight than CategorySplitted
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchCategoryWeight : LuceneConfig.WildcardsSearchCategoryWeight;
+                        LuceneConfig.SearchCategoryWeight : (LuceneConfig.WildcardsSearchCategoryWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.FullCategoryName):
-                    query.Boost = isWildcard == false?
-                        LuceneConfig.SearchCategoryWeight : LuceneConfig.WildcardsSearchCategoryWeight;
+                    query.Boost = isWildcard == false ?
+                        LuceneConfig.SearchCategoryWeight : (LuceneConfig.WildcardsSearchCategoryWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.CategorySplitted):
                     //Under this case the CategorySplitted field will have more weight than NameSplitted
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchNameWeight : LuceneConfig.WildcardsSearchNameWeight;
+                        LuceneConfig.SearchNameWeight : (LuceneConfig.WildcardsSearchNameWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.Description):
                     query.Boost = isWildcard == false ?
-                        LuceneConfig.SearchDescriptionWeight : LuceneConfig.WildcardsSearchDescriptionWeight;
+                        LuceneConfig.SearchDescriptionWeight : (LuceneConfig.WildcardsSearchDescriptionWeight - boostOffset);
                     break;
                 case nameof(LuceneConfig.NodeFieldsEnum.SearchKeywords):
                     query.Boost = isWildcard == false ?
-                       LuceneConfig.SearchTagsWeight : LuceneConfig.WildcardsSearchTagsWeight;
+                       LuceneConfig.SearchTagsWeight : (LuceneConfig.WildcardsSearchTagsWeight - boostOffset);
+                    break;
+                case nameof(LuceneConfig.NodeFieldsEnum.Parameters):
+                    query.Boost = 1;
                     break;
                 default:
                     query.Boost = isWildcard == false ?
-                       LuceneConfig.SearchMetaFieldsWeight : LuceneConfig.WildcardsSearchMetaFieldsWeight;
+                       LuceneConfig.SearchMetaFieldsWeight : (LuceneConfig.WildcardsSearchMetaFieldsWeight - boostOffset);
                     break;
             }
+
+            //This section only applies if the SearchTerm contain empty spaces then is splitted by empty space and try to calculate the weight for each part
```

Comment:
The comment can be improved here, it is a bit confusing
Do you mean the section only applies to the search term which were splitted before?

Reply:
I've updated the comment for making it more clear please let me know if is ok, see the commit: https://github.com/DynamoDS/Dynamo/pull/15473/commits/40697ac1b0752b02a53c5ccb86b77a1a67747492